mlibrary/my_account
# Updating ILLiad requests and specs

# Overview
Ralph has provided us with ILLiad test data so we can make better use of the API.
```
TN 3298017
Loan
Title: Test Book One
Status: Request Finished

TN 3298019
Loan
Title: Another Test Book
Status: Checked Out to Customer

TN 3298020
Article
Article Title: What I Think
Status: Request Finished

TN 3298028
Book Chapter
Title: A Book About Things
Status: Request Finished

TN 3298029
Article
Article Title: Who Am I?
Status: Delivered to Web
```

`./spec/fixtures/illiad_requests.json` has been updated with the new information. The tests and model have been appropriately updated to apply this new information.

## Future Changes
* We should start pulling ILLiad API information directly from their site.
* Update the API information to pull only current information (`GET /ILLiadWebPlatform/Transaction/UserRequests/testhelp?$filter=TransactionStatus eq 'Request Sent' HTTP/1.1`).
* Update the table information in From Other Institutions (Interlibrary Loan) to match what shows on ILLiad.

## Testing
* Run the tests to make sure they pass (`docker-compose run web bundle exec rspec`). Break the tests to make sure they're working.
* Check [From Other Institutions (Interlibrary Loan)](http://localhost:4567/requests/interlibrary-loan) to see if one request shows up.
* Check the rest of the site to make sure nothing is broken.
